source/SH3/system/msg/msg.cpp

```diff
@@ -0,0 +1,40 @@
+/** @file
+ *  Implementation of msg.hpp
+ *
+ *
+ *  @copyright 2016-2017  Palm Studios
+ *
+ *  @date 16-1-2017
+ *
+ *  @author Jesse Buhagiar
+ */
+#include "SH3/SH3/system/msg/msg.hpp"
+
+using namespace sh3_system;
+
+void sh3_msg::Load(sh3_arc_vfile& file)
+{
+
+    std::uint8_t c;
+    std::uint8_t c2;
+    sh3_arc_vfile::read_error e;
+
```

Comment:
We should `msgString.reserve()` an approximate message length, by adding an accessor for `sh3_arc_vfile::fsize`.
`msgString.shrink_to_fit()` can be used to free any excess memory.